services/rfq/api/rest/rfq.go

```diff
@@ -0,0 +1,213 @@
+			}
+
+			// record the response
+			err = r.db.InsertActiveQuoteResponse(collectionCtx, resp, respStatus)
+			if err != nil {
+				logger.Errorf("Error inserting active quote response: %v", err)
+			}
+		}(client)
+		return true
+	})
+
+	// wait for all responses to be received, or expiration
+	select {
+	case <-expireCtx.Done():
+		// request expired before all responses were received
+	case <-func() chan struct{} {
+		ch := make(chan struct{})
+		go func() {
+			wg.Wait()
+			close(ch)
+		}()
+		return ch
+	}():
+		// all responses received
+	}
+
+	return responses
+}
+
+func getBestQuote(a, b *model.QuoteData) (*model.QuoteData, bool) {
+	if a == nil && b == nil {
+		return nil, false
+	}
+	if a == nil {
+		return b, true
+	}
+	if b == nil {
+		return a, false
+	}
+	aAmount, _ := new(big.Int).SetString(*a.DestAmount, 10)
+	bAmount, _ := new(big.Int).SetString(*b.DestAmount, 10)
+	if aAmount.Cmp(bAmount) > 0 {
+		return a, false
+	}
+	return b, true
+}
+
+func getQuoteResponseStatus(ctx context.Context, resp *model.RelayerWsQuoteResponse, relayerAddr string) db.ActiveQuoteResponseStatus {
+	respStatus := db.Considered
+	err := validateRelayerQuoteResponse(relayerAddr, resp)
+	if err != nil {
+		respStatus = db.Malformed
+		logger.Errorf("Error validating quote response: %v", err)
+	} else if ctx.Err() != nil {
+		respStatus = db.PastExpiration
+	}
+	return respStatus
+}
+
+func validateRelayerQuoteResponse(relayerAddr string, resp *model.RelayerWsQuoteResponse) error {
+	if resp.Data.RelayerAddress == nil {
+		return fmt.Errorf("relayer address is nil")
+	}
+	// TODO: compute quote ID from request
+	resp.QuoteID = uuid.New().String()
+	resp.Data.RelayerAddress = &relayerAddr
+	return nil
+}
+
+func (r *QuoterAPIServer) recordActiveQuote(ctx context.Context, quote *model.QuoteData, requestID, quoteID string) (err error) {
+	if quote == nil {
+		err = r.db.UpdateActiveQuoteRequestStatus(ctx, requestID, nil, db.Expired)
+		if err != nil {
+			logger.Errorf("Error updating active quote request status: %v", err)
+		}
+	} else {
+		err = r.db.UpdateActiveQuoteRequestStatus(ctx, requestID, &quoteID, db.Fulfilled)
+		if err != nil {
+			logger.Errorf("Error updating active quote request status: %v", err)
+		}
+		err = r.db.UpdateActiveQuoteResponseStatus(ctx, quoteID, db.Returned)
+		if err != nil {
+			return fmt.Errorf("error updating active quote response status: %w", err)
+		}
+	}
+	return nil
+}
+
+func (r *QuoterAPIServer) handlePassiveRFQ(ctx context.Context, request *model.PutUserQuoteRequest) (*model.QuoteData, error) {
+	quotes, err := r.db.GetQuotesByOriginAndDestination(ctx, uint64(request.Data.OriginChainID), request.Data.OriginTokenAddr, uint64(request.Data.DestChainID), request.Data.DestTokenAddr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get quotes: %w", err)
+	}
+
+	originAmount, ok := new(big.Int).SetString(request.Data.OriginAmount, 10)
+	if !ok {
+		return nil, fmt.Errorf("invalid origin amount")
+	}
```

Comment:
[nit] if you're not wrapping an error, prefer errors.New()

dd961c19e4a4f9fdd820289c8a6e5c587719c53f 

before for comparison:

<img width="601" alt="image" src="https://github.com/user-attachments/assets/1acdde5e-38c1-4595-9871-f23665fd7528">